packages/@aws-cdk/aws-rds/lib/proxy.ts

```diff
@@ -0,0 +1,429 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import * as cdk from '@aws-cdk/core';
+import { IDatabaseCluster } from './cluster-ref';
+import { IDatabaseInstance } from './instance';
+import { CfnDBProxy, CfnDBProxyTargetGroup } from './rds.generated';
+
+/**
+ * The kinds of databases that the proxy can connect to.
+ * This value determines which database network protocol the proxy recognizes when it interprets network traffic to
+ * and from the database.
+ * The engine family applies to MySQL and PostgreSQL for both RDS and Aurora.
+ */
+export enum ProxyEngineFamily {
+  /**
+   * MYSQL
+   */
+  MYSQL = 'MYSQL',
+
+  /**
+   * POSTGRESQL
+   */
+  POSTGRESQL = 'POSTGRESQL',
+}
+
+/**
+ * SessionPinningFilter
+ *
+ * @see https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/rds-proxy.html#rds-proxy-pinning
+ */
+export enum SessionPinningFilter {
+  /**
+   * You can opt out of session pinning for the following kinds of application statements:
+   *
+   * - Setting session variables and configuration settings.
+   */
+  EXCLUDE_VARIABLE_SETS = 'EXCLUDE_VARIABLE_SETS'
+}
+
+/**
+ * Specifies the settings that control the size and behavior of the connection pool.
+ */
+export interface ProxyConnectionPool {
```

Comment:
This type is directly related to the type `CfnDBProxyTargetGroup.ConnectionPoolConfigurationInfoFormatProperty` and the section on the AWS Management Console, so that users can easily understand what the props mean.

![image](https://user-images.githubusercontent.com/15084045/85604796-93d35a00-b68c-11ea-95a6-167384267a9f.png)

Reply:
That's not how the CDK's ergonomics work. Since the CDK is also available in other languages that use nominal typing such as Java, it becomes more painful to construct another object just to add another set of configurations.

Let's merge them in, as I have suggested.